engines/router/missionctl/server/upi/server.go

```diff
@@ -40,9 +37,67 @@ func NewUPIServer(mc missionctl.MissionControlUPI) *Server {
 	}
 }
 
+// UniversalPredictionServiceServer is the internal server that communicates in bytes
+// It requires the codec to be passing []byte instead of a proto object. It is registered
+// the same way as the generated proto which has the following interface instead.
+// PredictValues(context.Context, *PredictValuesRequest) (*PredictValuesResponse, error)
+type UniversalPredictionServiceServer interface {
+	PredictValues(context.Context, []byte) ([]byte, error)
+}
+
+//nolint:revive,gofmt,goimports
+// UniversalPredictionServiceServiceDesc implement grpc internal methodHandler interface, it creates an io.writer
+// and expect the codec to write bytes into the input instead of the typical proto request.
+// Nolint is used as context cannot be changed to first argument in order to fit the interface and there are
+// other bugs on fmt and imports while using nolint
+func UniversalPredictionServiceServiceDesc(
+	srv interface{},
+	ctx context.Context,
+	dec func(interface{}) error,
+	interceptor grpc.UnaryServerInterceptor,
+) (interface{}, error) {
+	in := new(bytes.Buffer)
+	if err := dec(in); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return srv.(UniversalPredictionServiceServer).PredictValues(ctx, in.Bytes())
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: "/caraml.upi.v1.UniversalPredictionService/PredictValues",
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(UniversalPredictionServiceServer).PredictValues(ctx, req.([]byte))
```

Comment:
Since it is the fiber codec being used, input `[]byte` will be passed as it is to `PredictValues`

Likewise when sending the response, `[]byte` is sent and 'marshal` by the codec to remain as `[]byte`